controllers/events/event.go

```diff
@@ -172,25 +172,38 @@ func newSinkComponentSpec(c client.Client, log logr.Logger, ref *ofevent.Referen
 }
 
 func createSinkComponent(ctx context.Context, c client.Client, log logr.Logger, resource client.Object, sink *ofevent.SinkSpec) (*componentsv1alpha1.Component, error) {
+	if sink.Uri == nil && sink.Ref == nil {
+		return nil, errors.New("at least one uri or ref must be set in sink")
+	}
+
+	var namespace, url string
+	if sink.Uri != nil {
+		url = *sink.Uri
+		// when setting the Uri, use resource.GetNamespace()
+		namespace = resource.GetNamespace()
+	} else {
+		var ksvc kservingv1.Service
+		if err := c.Get(ctx, types.NamespacedName{Namespace: sink.Ref.Namespace, Name: sink.Ref.Name}, &ksvc); err != nil {
+			log.Error(err, "Failed to find Knative Service", "namespace", sink.Ref.Namespace, "name", sink.Ref.Name)
+			return nil, err
+		}
+		url = ksvc.Status.URL.String()
+		namespace = sink.Ref.Namespace
+	}
```

Comment:
I think we also need to add support for Function in this section, what do you think?

Reply:
```yaml
apiVersion: core.openfunction.io/v1alpha2
kind: Function
name: function-sample
namespace: default
```

like this ?

Reply:
yes